content/en/docs/Reference/veneers.md

```diff
@@ -0,0 +1,397 @@
+---
+title: "Veneers"
+linkTitle: "Veneers"
+weight: 5
+date: 2022-06-30
+---
+
+File-Based Catalogs (FBC) are a major improvement to the imperative update graph approaches of previous versions. FBCs give operator authors a [declarative and deterministic approach to defining their update graph](https://olm.operatorframework.io/docs/concepts/olm-architecture/operator-catalog/creating-an-update-graph/). However, FBCs can get complex, especially as the number of releases and dependencies scale.
+
+Enter the `veneer`, a simpler transitional step to generating an FBC.
+
+## Design
+
+There are two components to every `veneer`:
+1. An arbitrary API
+2. An executable which processes #1 and produces a valid FBC.
+
+## Benefits
+
+- A set of consistent expectations from including as standard functionality with `opm`
+- A good set of use-cases for simplifying the ability of operator authors to manage FBCs, including contributing them to an index
+- A transitive benefit from consistency of practice as authors move from operator to operator
+
+The veneers which accompany `opm` are:
+- [`basic veneer`](#basic-veneer), which provides simplification of FBC concepts; and
+- [`semver veneer`](#semver-veneer), which provides the capability of sophisticated channel generation
+
+## Basic Veneer
+The `basic veneer` is an input schema which eliminates FBC information that can be retrieved from existing registries when we process it.
+Users provide all components of an [FBC schema](https://olm.operatorframework.io/docs/reference/file-based-catalogs/#olm-defined-schemas), but supply only the bundle image reference for any `olm.bundle` objects. This results in a greatly simplified, smaller document.
+
+
+In a very simple example, we define an `olm.package` and an `olm.channel` composed of two `olm.bundle` objects that have an image name attribute but no other attributes/properties.
+```yaml
```

Comment:
@grokspawn @michaelryanpeter I think we need something like here: 

```suggestion

### When is recommended for index catalog admins to adopt this veener

- If you are **not** looking to ensure standards and the content that will be distributed in your catalog
- if you are OK with Operator Authors defining channel names which make more sense for their own specific product instead of having a common nomenclature which can help cluster admins better understand the maturity of the solutions distributed in
- If you are not looking to ensure that all Operator solutions are distributed in the index following semver

This option is **not** recommended for those who are versioning the indexes and allowing Operator Authors to provide solutions to be distributed for each index version or those that are looking for common terminologies and standards on what is distributed in the catalog. If that is your case, please check the seemver veener, 

```yaml
```

Reply:
I'm not sure I agree with this sentiment. If all of this was actually true, what value would there be in even creating the simple veneer in the first place? I think the purpose of these tools need to be clearly defined, but I'm skeptical that we should ever be writing documentation like this from the perspective of a superset catalog maintainer / pipeline developer that isn't directly interacting with the content of these catalogs.

Reply:
Hi @kevinrizza,  

- Operators Authors produce Operators to abstract the complexities for cluster admins
- Index Catalog admins manage/built index to distribute the content on the cluster for cluster admins.
Therefore, all solutions and roles exist to address an expected result for cluster admins. 

In this way, for we choose what is the best we need to understand what problems each case scenario can solve for cluster admins and the real motivations. Otherwise, we cannot adopt A or B just because is easier to build. 

Operator Authors should NOT design solutions and make choices based on it is easy to get done, they should do something that will provide a better result for those that will consume their solutions integrated with OLM (cluster admins). 

Therefore, those that will try to validate and ensure the quality of their solution are those that (IMHO) will decide if they want to accept content made with the veener A or B. Those will also opt-in or not based on what promotes the best result for cluster admins. 

In this way, if we do not provide what are the benefits to adopt one case or another they cannot choose properly, then we will not have the best results at the end goal.